crates/re_log_types/src/path/parse_path.rs

```diff
@@ -22,25 +29,243 @@ pub enum PathParseError {
+                } else {
+                    return Err(PathParseError::BadInstanceKey(
+                        instance_key_token.to_owned(),
+                    ));
+                }
+            } else {
+                return Err(PathParseError::BadInstanceKey(
+                    instance_key_token.to_owned(),
+                ));
+            }
+            tokens.truncate(bracket);
+        }
+
+        // The remaining tokens should all be separated with `/`:
+
+        let entity_path_parts = entity_path_parts_from_tokens(&tokens)?;
+        let entity_path = EntityPath::from(entity_path_parts);
+
+        Ok(Self {
+            entity_path,
+            instance_key,
+            component_name,
+        })
+    }
+}
+
+impl FromStr for EntityPath {
+    type Err = PathParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let DataPath {
+            entity_path,
+            instance_key,
+            component_name,
+        } = DataPath::from_str(s)?;
+
+        if let Some(instance_key) = instance_key {
+            return Err(PathParseError::UnexpectedInstanceKey(instance_key));
+        }
+        if let Some(component_name) = component_name {
+            return Err(PathParseError::UnexpectedComponentName(component_name));
+        }
+
+        Ok(entity_path)
     }
+}
+
+impl FromStr for ComponentPath {
+    type Err = PathParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let DataPath {
+            entity_path,
+            instance_key,
+            component_name,
+        } = DataPath::from_str(s)?;
 
-    if path == "/" {
+        if let Some(instance_key) = instance_key {
+            return Err(PathParseError::UnexpectedInstanceKey(instance_key));
+        }
+
+        let Some(component_name) = component_name else {
+            return Err(PathParseError::MissingComponentName);
+        };
+
+        Ok(ComponentPath {
+            entity_path,
+            component_name,
+        })
+    }
+}
+
+#[test]
+fn test_parse_component_path() {
+    assert_eq!(
+        ComponentPath::from_str("world/points.rerun.components.Color"),
+        Ok(ComponentPath {
+            entity_path: EntityPath::from_str("world/points").unwrap(),
+            component_name: "rerun.components.Color".into(),
+        })
+    );
+    assert_eq!(
+        ComponentPath::from_str("world/points.Color"),
+        Ok(ComponentPath {
+            entity_path: EntityPath::from_str("world/points").unwrap(),
+            component_name: "rerun.components.Color".into(),
+        })
+    );
+    assert_eq!(
+        ComponentPath::from_str("world/points.my.custom.color"),
+        Ok(ComponentPath {
+            entity_path: EntityPath::from_str("world/points").unwrap(),
+            component_name: "my.custom.color".into(),
+        })
+    );
+    assert_eq!(
+        ComponentPath::from_str("world/points."),
+        Err(PathParseError::TrailingDot)
+    );
+}
```

Comment:
Missing an error case for component instance path and missing component part
* `world/points[#42].rerun.components.Color` should fail with `PathParseError::UnexpectedInstanceKey`
* `world/points` should fail with `PathParseError::MissingComponentName`

Reply:
also, similar test for `EntityPath` would be nice :)

Reply:
`EntityPath` is tested at the end of the file

Reply:
thanks!